### PR TITLE
perf_pcp.py:Updated code for fetching perfevent_24x7_events.

### PR DESCRIPTION
**Pull request by naveen538**
Recently there are changes in hv_24x7_events
So, updated code for fetching PM_MBA0_CLK_CYC and PM_PB_CYC events.

Signed-off-by: Naveen kumar T <naveet89@in.ibm.com>

### CODE REVIEW

**Comment by naveen538** [comment]
Before Patch
-----------------
avocado run perf_pcp.py
JOB ID     : 26c7303cf409dc0f0b0c6a95aebff7f198de5ce0
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2020-05-28T11.15-26c7303/job.log
 (1/5) perf_pcp.py:PCP.test_pmcd_daemon: PASS (48.20 s)
 (2/5) perf_pcp.py:PCP.test_pmda_perfevent_install: PASS (5.42 s)
 (3/5) perf_pcp.py:PCP.test_pcp_cmd: PASS (0.86 s)
 (4/5) perf_pcp.py:PCP.test_perfevent_24x7_events: ERROR: Command 'pmval -s 2 perfevent.hwcounters.hv_24x7.PM_MBA0_CLK_CYC.value' failed.\nstdout: b''\nstderr: b'pmval: pmLookupName(perfevent.hwcounters.hv_24x7.PM_MBA0_CLK_CYC.value): Unknown metric name\n'\nadditional_info: None (0.84 s)
 (5/5) perf_pcp.py:PCP.test_pmda_perfevent_remove: PASS (4.04 s)
RESULTS    : PASS 4 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 59.55 s
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2020-05-28T11.15-26c7303/results.html


After Patch
--------------
avocado run perf_pcp.py
JOB ID     : f013ec2e88ea8baad63e19d723a228348c9928c4
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2020-05-28T11.28-f013ec2/job.log
 (1/5) perf_pcp.py:PCP.test_pmcd_daemon: PASS (2.29 s)
 (2/5) perf_pcp.py:PCP.test_pmda_perfevent_install: PASS (5.53 s)
 (3/5) perf_pcp.py:PCP.test_pcp_cmd: PASS (0.86 s)
 (4/5) perf_pcp.py:PCP.test_perfevent_24x7_events: PASS (4.83 s)
 (5/5) perf_pcp.py:PCP.test_pmda_perfevent_remove: PASS (4.05 s)
RESULTS    : PASS 5 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 17.80 s
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2020-05-28T11.28-f013ec2/results.html

File: perf/perf_pcp.py [hunk 1]
```diff
@@ -90,10 +90,10 @@ def test_perfevent_24x7_events(self):
         if self.cpu_family == 'power8':
             self.cancel("Not supported on Power8")
         # -s 2 collects two samples
-        cmd = "pmval -s 2 perfevent.hwcounters.hv_24x7.PM_MBA0_CLK_CYC.value"
+        cmd = "pmval -s 2 perfevent.hwcounters.hv_24x7.PM_MBA0_CLK_CYC_chip_0.value"
```
Comment (PraveenPenguin) [inline perf/perf_pcp.py:93]: just a thought can we maintain this as yaml input

**Comment by harish-24** [comment]
@naveen538 Commit message does not clearly state whether the change still works with older environment and whether that needs to be handled as well?

File: perf/perf_pcp.py [hunk 2]
```diff
@@ -89,12 +89,14 @@ def test_perfevent_24x7_events(self):
...
-        # -s 2 collects two samples
-        cmd = "pmval -s 2 perfevent.hwcounters.hv_24x7.PM_PB_CYC.value"
-        process.run(cmd, shell=True)
+        output = process.system_output("pminfo").decode()
+        for line in output.split('\n'):
+            # -s 2 collects two samples
+            if 'PM_MBA0_CLK_CYC' in line and 'value' in line:
+                process.run('pmval -s 2 ' + line, shell=True)
+            # -s 2 collects two samples
+            if 'PM_PB_CYC' in line and 'value' in line:
```
Comment (harish-24) [inline perf/perf_pcp.py:98]: same here

**Comment by naveen538** [comment]
checked with updated code
---------------------------------
avocado run perf_pcp.py
JOB ID     : 2c2042a8796e9203b94ad6688a8fde09e989997f
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2020-07-09T20.19-2c2042a/job.log
 (1/5) perf_pcp.py:PCP.test_pmcd_daemon: PASS (0.78 s)
 (2/5) perf_pcp.py:PCP.test_pmda_perfevent_install: PASS (5.15 s)
 (3/5) perf_pcp.py:PCP.test_pcp_cmd: PASS (0.90 s)
 (4/5) perf_pcp.py:PCP.test_perfevent_24x7_events: PASS (7.49 s)
 (5/5) perf_pcp.py:PCP.test_pmda_perfevent_remove: PASS (3.80 s)
RESULTS    : PASS 5 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2020-07-09T20.19-2c2042a/results.html
JOB TIME   : 18.54 s

File: perf/perf_pcp.py [hunk 3]
```diff
@@ -89,12 +89,14 @@ def test_perfevent_24x7_events(self):
...
+                process.run('pmval -s 2 ' + line, shell=True)
```
Comment (harish-24) [inline perf/perf_pcp.py:99]: `pmval -s 2 %s' % line` would be better.

Comment (harish-24) [inline perf/perf_pcp.py:99]: If you are sure `PM_PB_CYC` line will follow after `PM_MBA0_CLK_CYC` line, you can add a `break` next to this line so that unnecessary looping can be avoided.

File: perf/perf_pcp.py [hunk 4]
```diff
@@ -89,12 +89,14 @@ def test_perfevent_24x7_events(self):
...
-        # -s 2 collects two samples
-        cmd = "pmval -s 2 perfevent.hwcounters.hv_24x7.PM_MBA0_CLK_CYC.value"
-        process.run(cmd, shell=True)
-        # -s 2 collects two samples
-        cmd = "pmval -s 2 perfevent.hwcounters.hv_24x7.PM_PB_CYC.value"
-        process.run(cmd, shell=True)
+        output = process.system_output("pminfo").decode()
+        for line in output.split('\n'):
+            # -s 2 collects two samples
+            if 'PM_MBA0_CLK_CYC' in line and 'value' in line:
```
[the remark this answers is not in the document]
Comment (naveen538) [reply perf/perf_pcp.py:94]: Not able to fetch exact output from : if ("PM_MBA0_CLK_CYC" and "value") in line:

>>> for line in event.splitlines():
...  if ("PM_MBA0_CLK_CYC" and "value") in line:
...   print line
... 
perfevent.hwcounters.hv_24x7.PM_MBA5_WRITE_BYTES.value
perfevent.hwcounters.hv_24x7.PM_MBA7_WRITE_BYTES.value
perfevent.hwcounters.hv_24x7.PM_MBA7_READ_BYTES.value
perfevent.hwcounters.hv_24x7.PM_MBA0_CLK_CYC_chip_0.value
perfevent.hwcounters.hv_24x7.PM_MBA2_READ_BYTES.value
perfevent.hwcounters.hv_24x7.PM_MBA7_CLK_CYC.value
perfevent.hwcounters.hv_24x7.PM_MBA5_CLK_CYC.value


>>> for line in event.splitlines():
...  if 'PM_MBA0_CLK_CYC' in line and 'value' in line:
...   print line
... 
perfevent.hwcounters.hv_24x7.PM_MBA0_CLK_CYC_chip_0.value
>>>

[the remark this answers is not in the document]
Comment (harish-24) [reply perf/perf_pcp.py:95]: I guess `if 'PM_MBA0_CLK_CYC' in line and 'value' in line:` and `if ("PM_MBA0_CLK_CYC" and "value") in line:` works the same way? Can you paste the difference b/w these two clearly?

File: perf/perf_pcp.py [hunk 5]
```diff
@@ -89,12 +89,16 @@ def test_perfevent_24x7_events(self):
...
-        process.run(cmd, shell=True)
-        # -s 2 collects two samples
-        cmd = "pmval -s 2 perfevent.hwcounters.hv_24x7.PM_PB_CYC.value"
-        process.run(cmd, shell=True)
+        output = process.system_output("pminfo").decode()
+        for line in output.splitlines():
+            # -s 2 collects two samples
+            if 'PM_MBA0_CLK_CYC' in line and 'value' in line:
+                process.run('pmval -s 2 %s' % line, shell=True)
+                break
```
Comment (harish-24) [inline perf/perf_pcp.py:97]: Why is this added? It never reaches the below code in loop with this added.

Comment (naveen538) [reply perf/perf_pcp.py]: Removed break...

File: perf/perf_pcp.py (hunk 1 above)
Comment (naveen538) [reply perf/perf_pcp.py]: As this events may have different values in different systems , so modifying code to fetch events dynamically .

**Comment by harish-24** [comment]
@naveen538 While we wait for @PraveenPenguin to review, can you paste a run result with recent changes.

**Comment by PraveenPenguin** [comment]
@naveen538 can you please attach  job log, please

**Comment by naveen538** [comment]
Job log with recent changes
------------------------------------
avocado run perf_pcp.py

JOB ID     : 37786e6b9de2de03de6eff0580d316915e4bba96
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2020-07-15T15.12-37786e6/job.log
 (1/5) perf_pcp.py:PCP.test_pmcd_daemon: PASS (0.79 s)
 (2/5) perf_pcp.py:PCP.test_pmda_perfevent_install: PASS (5.13 s)
 (3/5) perf_pcp.py:PCP.test_pcp_cmd: PASS (0.93 s)
 (4/5) perf_pcp.py:PCP.test_perfevent_24x7_events: PASS (3.41 s)
 (5/5) perf_pcp.py:PCP.test_pmda_perfevent_remove: PASS (4.24 s)
RESULTS    : PASS 5 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2020-07-15T15.12-37786e6/results.html
JOB TIME   : 14.86 s
